### ISSUE TITLE
P1001 Can't connect to MySQL/MariaDB with sql_mode set to ANSI or ANSI_QUOTES

### ISSUE THREAD

**Issue by vidarinho** [prisma/prisma#17112]
### Bug Description
**tldr;** setting `sql_mode=...,ANSI_QUOTES,...` in MySQL/MariaDB gives the error: `P1001 Can't reach database server at <HOST>:3306` even if the database is available.

Ok, so this is an obscure one, I was able to solve it by not setting ANSI or ANSI_QUOTES in sql_mode, but I will leave this here for future reference for someone else which has this issue. Also if the engine team can take a look as according to DBA non ANSI mode can have unwanted consequences.

```
$ DEBUG="*" npx prisma db push
  prisma:engines  binaries to download libquery-engine, migration-engine, introspection-engine, prisma-fmt +0ms
  prisma:loadEnv  project root found at /app/package.json +0ms
  prisma:tryLoadEnv  Environment variables not found at null +0ms
  prisma:tryLoadEnv  Environment variables not found at undefined +1ms
  prisma:tryLoadEnv  No Environment variables loaded +0ms
  prisma:getConfig  Using getConfig Wasm +0ms
  prisma:getConfig  config data retrieved without errors in getConfig Wasm +16ms
  prisma:loadEnv  project root found at /app/package.json +21ms
  prisma:tryLoadEnv  Environment variables not found at null +22ms
  prisma:tryLoadEnv  Environment variables not found at undefined +1ms
  prisma:tryLoadEnv  No Environment variables loaded +2ms
Prisma schema loaded from db/mysql-schema.prisma
  prisma:getConfig  Using getConfig Wasm +23ms
  prisma:getConfig  config data retrieved without errors in getConfig Wasm +5ms
Datasource "db": MySQL database "<DB NAME>" at "<HOST>:3306"
  prisma:getConfig  Using getConfig Wasm +25ms
  prisma:getConfig  config data retrieved without errors in getConfig Wasm +4ms
  prisma:getConfig  Using getConfig Wasm +10ms
  prisma:getConfig  config data retrieved without errors in getConfig Wasm +7ms

Error: Error: P1001: Can't reach database server at `<HOST>`:`3306`

Please make sure your database server is running at `<HOST>`:`3306`.
    at ensureDatabaseExists (/app/node_modules/prisma/build/index.js:93119:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async _DbPush.parse (/app/node_modules/prisma/build/index.js:94521:28)
    at async main (/app/node_modules/prisma/build/index.js:98674:18)
```

The database can be connected to via mysql-client or any other database client I tested.
I was able to isolate the problem by looking at what command the prisma script ran. The error message comes regardless of the user name and password entered.

```
$ ./node_modules/@prisma/engines/migration-engine-linux-musl-openssl-3.0.x cli --datasource mysql://U:P@<HOST>/db can-connect-to-database

{"timestamp":"2023-01-03T10:35:24.622493039Z","level":"ERROR","fields":{"is_panic":false,"error_code":"P1001","message":"Can't reach database server at `<HOST>`:`3306`\n\nPlease make sure your database server is running at `<HOST>`:`3306`."},"target":"migration_engine::logger"}
```

The problem did not occur when running MariaDb in a "clean" docker container from Docker hub, so I started to investigate the differences in config from the Docker hub version and our DB setup at our hosting provider, which lead me to the sql_mode setting in MySQL config.

```
# FAILS (DB setup at hosting provider)
sql_mode= REAL_AS_FLOAT,PIPES_AS_CONCAT,ANSI_QUOTES,IGNORE_SPACE,ANSI,STRICT_TRANS_TABLES

# WORKS (clean MariaDB on Docker hub)
sql_mode=IGNORE_SPACE,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
```

### Environment & setup
Prisma version: 4.6, 4.8
SQL version MariaDb: 10.6.11 (MySQL not tested)
Node version: 16,19
Client OS: Mac, Linux

**Comment by sacru2red** [comment]
i have same issue

**Comment by BrianValente** [comment]
Same issue here! With MariaDB, I need ANSI_QUOTES for other microservices.
